src/kinetics/InterfaceKinetics.cpp

```diff
@@ -656,6 +695,99 @@ SurfaceArrhenius InterfaceKinetics::buildSurfaceArrhenius(
     return rate;
 }
 
+BMSurfaceArrhenius InterfaceKinetics::buildBMSurfaceArrhenius(
+    size_t i, BMInterfaceReaction& r, bool replace)
+{
+    if (r. is_sticking_coefficient) {
+        // Identify the interface phase
```

Comment:
Almost all of this method is duplicating the contents of `buildSurfaceArrhenius`. Is there a way to refactor this code to avoid the duplication?